lib/ts/recipe/session/components/themes/accessDeniedScreenTheme/index.tsx

```diff
@@ -1,28 +1,50 @@
 /* eslint-disable react/jsx-no-literals */
-import React, { Fragment } from "react";
+import React from "react";
 
+// import { signOut } from "../../..";
 import { withOverride } from "../../../../../components/componentOverride/withOverride";
 import { hasFontDefined } from "../../../../../styles/styles";
+import SuperTokens from "../../../../../superTokens";
 import { useTranslation } from "../../../../../translation/translationContext";
+import { useUserContext } from "../../../../../usercontext";
+import Session from "../../../recipe";
+import { BackButton } from "../../library/backButton";
+import { LogoutButton } from "../../library/logoutButton";
 import { ThemeBase } from "../themeBase";
 
 import type { AccessDeniedThemeProps } from "../../../types";
 import type { FC } from "react";
 
-const AccessDeniedScreen: FC<AccessDeniedThemeProps> = (props) => {
+const AccessDeniedScreen: FC<AccessDeniedThemeProps> = () => {
+    const userContext = useUserContext();
     const t = useTranslation();
-    const { validatorId, reason = {} } = props.denialInfo || {};
-    const failureReason =
-        typeof reason === "string" ? reason : typeof reason.message === "string" ? reason.message : null;
+    const history = SuperTokens.getReactRouterDomWithCustomHistory()?.useHistoryCustom();
+
+    const onLogout = async () => {
+        try {
+            await Session.getInstanceOrThrow().signOut({ userContext });
+        } catch (error) {
+        } finally {
+            history.navigate("/auth");
+        }
+    };
+
+    const onBackButtonClicked = () => {
+        return SuperTokens.getInstanceOrThrow().redirectToAuth({
+            show: "signin",
+            history,
+            redirectBack: false,
+        });
+    };
```

Comment:
This isn't really a back button. What's the best we can do to emulate the "go back" button of the browser? Or would that just get us to the page that redirected us to this page in case it was a validation failure in `SessionAuth`?